src/lib/Scenes/Home/Components/ForYou/index.tsx

```diff
@@ -15,80 +15,124 @@ interface DataSourceRow {
   }
 
   componentDidMount() {
-    const rows: DataSourceRow[] = []
-    const artworkModules = this.props.forYou.artwork_modules || []
-    const artistModules = this.props.forYou.artist_modules && this.props.forYou.artist_modules.concat()
-    const fairsModule = this.props.forYou.fairs_module
-    rows.push({ type: "fairs", data: fairsModule })
-    for (let i = 0; i < artworkModules.length; i++) {
-      const artworkModule = artworkModules[i]
-      rows.push({ type: "artwork", data: artworkModule })
-      if ((i + 1) % 2 === 0) {
+    const { forYou } = this.props
+    const rowData = []
+    const artworkModules = forYou.artwork_modules || []
+    const artistModules = forYou.artist_modules && forYou.artist_modules.concat()
+    const fairsModule = forYou.fairs_module
+
+    rowData.push({
+      type: "fairs",
+      data: fairsModule,
+    })
+
+    artworkModules.forEach((artworkModule, index) => {
+      rowData.push({
+        type: "artwork",
+        data: artworkModule,
+      })
+
+      const alternateRow = (index + 1) % 2 === 0
+
+      if (alternateRow) {
         const artistModule = artistModules.shift()
         if (artistModule) {
-          rows.push({ type: "artist", data: artistModule })
+          rowData.push({
+            type: "artist",
+            data: artistModule,
+          })
         }
       }
-    }
+    })
+
     this.setState({
-      dataSource: this.state.dataSource.cloneWithRows(rows),
+      rowData,
     })
   }
 
-  handleRefresh = () => {
-    this.setState({ isRefreshing: true })
-    const stopRefreshing = () => this.setState({ isRefreshing: false })
-
-    Promise.all(
-      this.state.modules.map(module => {
-        return new Promise((resolve, reject) => {
-          module.forceFetch(null, readyState => {
-            if (readyState.error) {
-              reject(readyState.error)
-            } else if (readyState.aborted) {
-              reject()
-            } else if (readyState.done) {
-              resolve()
-            }
+  handleRefresh = async () => {
+    this.setState({
+      isRefreshing: true,
+    })
+
+    try {
+      await Promise.all(
+        this.modules.map(module => {
+          return new Promise((resolve, reject) => {
+            const { props, state: { data, relayProp } } = module
+
+            relayProp.refetch({ ...props.rail, fetchContent: true }, null, error => {
+              if (error) {
+                console.error("Home/ForYou | ", error)
+                reject(error)
+              } else {
+                resolve()
+              }
+            })
           })
         })
+      )
+
+      // Success
+      this.setState({
+        isRefreshing: false,
       })
-    ).then(stopRefreshing, stopRefreshing)
+
+      // Fail
+    } catch (errors) {
+      this.setState({
+        errors, // TODO: Display this somehow
+      })
+    }
+  }
+
+  handleScroll = event => {
+    this.currentScrollOffset = event.nativeEvent.contentOffset.y
```

Comment:
I wonder if this is still for the 1px scroll hack we had at some point to work around a bug in RN, if so this can just be removed. /cc @sarahscott